Source/Urho3D/CSharp/Managed/Scene/Node.cs

```diff
@@ -34,9 +38,68 @@ public partial class Node
             return (T)GetComponent(typeof(T).Name, recursive);
         }
 
+        /// <summary>
+        /// Get first occurance of a component type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
         public T GetComponent<T>() where T : Component
         {
             return (T)GetComponent(typeof(T).Name);
         }
+
+        /// <summary>
+        /// get all components of a type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public ComponentList GetComponentsbyType<T>(bool recursive = false) where T : Component
+        {
+            ComponentList cplist = new ComponentList();
+            GetComponents(cplist, typeof(T).Name, recursive);
+            return cplist;
+        }
+
+        /// <summary>
+        /// Remove all Components of a specific Type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void RemoveAllComponents<T>() where T : Component
+        {
+            var childrenNode = GetComponentsbyType<T>();
+            foreach (var childnode in childrenNode)
+            {
+                RemoveComponent(childnode);
+            }
+            Debug.Assert(GetComponent<T>() is null, "Components are not removed");
+        }
+
+        /// <summary>
+        /// Remove all childern of a specific Type
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void RemoveAllChildrenbyName(string NodeName, bool recursive = false)
+        {
+            var childrenNode = GetChildren(recursive).Where(o => o.Name == NodeName);
+            foreach (var childnode in childrenNode)
+            {
+                RemoveChild(childnode);
+            }
+            Debug.Assert(GetChild(NodeName) is null, "Children are not removed");
+        }
+
+
+        /// <summary>
+        /// Save to XML File
+        /// </summary>
+        /// <param name="Path"></param>
+        /// <param name="RootNode"></param>
+        /// <returns></returns>
+        public bool SaveXML(string Path, string RootNode = "RootNode")
+        {
+            XMLFile file = new XMLFile(Context);
+            var fileroot = file.CreateRoot("RootNode");
+            return SaveXML(Path);
```

Comment:
`Serializable` has `LoadFile(string)` and `SaveFile(string)` which autodetects format by file extension of passed path. That does not work for you?

Reply:
well I didn't find it under `Serializable` . I only found it under Resources. so I assumed `Resources.SaveFile(string)` is related to resources only, isn't it?

Reply:
This is something little different. Seems like i only added `Serializable::LoadFile` as i needed it. And i forgot `SaveFile()`.

Reply:
so, do you see it is convenient to push this update or omit it for a better implementation?

Reply:
I will add `SaveFile()` to C++ class. There is no reason to have this available only for C# users ;)

Reply:
Or you could do it. `LoadFile()` is a good example on how to do it

Reply:
yep make sense... will implement